common/post-processing-worker-filter.h

```diff
@@ -21,9 +21,17 @@ class post_processing_worker_filter : public post_processing_filter
     {
     }
     ~post_processing_worker_filter()
+    {
+        release_background_worker();
```

Comment:
One thing we can do here is:
```
assert( ! _worker.joinable() );
```
What do you guys think? @ev-mp @nohayassin

Reply:
Agree, it is better than if condition

Reply:
returning if condition, it fails with assert